File: src/lib/models/slide.ts
export type TemplateView = "Song" | "Text" | "Media" | "Sermon" | "Scripture";

export interface MediaSlide {
  Media: {
    id: string
    type: "Video" | "Image"
    src: string
    template_id: number
  }
}

export interface ScriptureSlide {
  Scripture: {
    id: string
    type: "Scripture"
    text: string
    passage: string
    template_id: number
  }
}

export interface SermonSlide {
  Sermon: {
    id: string
    type: "Sermon"
    text: string
    template_id: number
  }
}

export interface SongSlide {
  Song: {
    id: string
    type: "Song"
    text: string
    verse: number
    template_id: number
  }
}

export interface TextSlide {
  Text: {
    id: string
    type: "Text"
    text: string
    template_id: number
  }
}

export type SlideType = MediaSlide | SermonSlide | ScriptureSlide | SongSlide | TextSlide

export interface View {
  id: string
  name: string
  type: TemplateView
  slide: SlideType
}

export type ScheduleType = View[]

File: src-tauri/src/app/models/slide.rs
use serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct MediaSlide {
    id: String,
    r#type: String,
    src: String,
    template_id: i32,
}

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct ScriptureSlide {
    id: String,
    r#type: String,
    text: String,
    passage: String,
    template_id: i32,
}

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct Alert {
    show: bool,
    r#type: String,
    title: String,
    content: String,
}

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct SermonSlide {
    id: String,
    r#type: String,
    text: String,
    template_id: i32,
}

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct SongSlide {
    id: String,
    r#type: String,
    text: String,
    verse: i32,
    template_id: i32,
}

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct TextSlide {
    id: String,
    r#type: String,
    text: String,
    template_id: i32,
}

#[derive(Serialize, Deserialize, Debug, Clone)]
pub enum SlideType {
    Media(MediaSlide),
    Sermon(SermonSlide),
    Scripture(ScriptureSlide),
    Song(SongSlide),
    Text(TextSlide),
}

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct View {
    id: String,
    name: String,
    r#type: String,
    slide: SlideType,
}

pub type ScheduleType = Vec<View>;
